src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -273,6 +280,13 @@ private String getBaseUrl(HttpServletRequest request) {
     return baseUrl;
   }
 
+  /**
+   * Gets the list of user-assigned categories from the request.
+   */
+  private Set<String> getCategories(HttpServletRequest request) {
+    return new HashSet<>(Arrays.asList(request.getParameterValues("categories")));
```

Comment:
I believe if there are duplicate categories, this will throw an exception. Is that what we want? If so, we may want to tell the user.

Reply:
I changed this method to use an ImmutableSet, so now it will remove duplicates. I have also added a unit test in #41 for duplicate categories.